### ISSUE TITLE
Purging data from Elasticsearch

### ISSUE THREAD

**Issue by buinauskas** [confluentinc/kafka-connect-elasticsearch#110]
As suggested by @rhauch on [SO](https://stackoverflow.com/questions/45576642/synchronizing-data-from-mssql-to-elasticsearch-using-apache-kafka?noredirect=1#comment78195309_45576642) it would be great if connector would be able to purge documents from Elasticsearch once it purged from topic.

**Comment by rhauch** [comment]
The suggestion is to handle tombstone sink records that have a non-null key and a null value. Tombstones are generally used to denote a deletion of a record, and thus could be translated into deletion requests for Elasticsearch. 

However, this would not be backward compatible, and as such the connector should _not do this by default_ but instead require explicitly enabling this feature via the connector's configuration properties.

**Comment by rhauch** [comment]
When this feature is not enabled, the connector should still accept but ignore tombstones.

**Comment by ewencp** [comment]
@rhauch Do `null` values work today or is the problem with `null`s only for keys? If they already cause things to break, we could reasonably include this as default behavior w/o adding configs since people couldn't be using the connector w/ them without already filtering those records out before they hit the connector.

**Comment by rhauch** [comment]
`null` values seem to fail no matter what. If the first record is a null before the schema can be inferred, you get this error:

```
org.apache.kafka.connect.errors.DataException: Cannot infer mapping without schema.
	at __randomizedtesting.SeedInfo.seed([4F1E066C58118185:4406F80B84A93F32]:0)
	at io.confluent.connect.elasticsearch.Mapping.inferMapping(Mapping.java:91)
	at io.confluent.connect.elasticsearch.Mapping.createMapping(Mapping.java:61)
	at io.confluent.connect.elasticsearch.ElasticsearchWriter.write(ElasticsearchWriter.java:201)
	at io.confluent.connect.elasticsearch.ElasticsearchSinkTask.put(ElasticsearchSinkTask.java:130)
```

If you can build the schema with non-null records and then get a null record, then you get this error when the connector tries to [construct a string](https://github.com/confluentinc/kafka-connect-elasticsearch/blob/master/src/main/java/io/confluent/connect/elasticsearch/DataConverter.java#L113) from a null `byte[]`:

```
java.lang.NullPointerException: null
	at __randomizedtesting.SeedInfo.seed([B6BB22D5E81D61FB:BDA3DCB234A5DF4C]:0)
	at java.lang.String.<init>(String.java:515)
	at io.confluent.connect.elasticsearch.DataConverter.convertRecord(DataConverter.java:99)
	at io.confluent.connect.elasticsearch.ElasticsearchWriter.write(ElasticsearchWriter.java:210)
	at io.confluent.connect.elasticsearch.ElasticsearchSinkTask.put(ElasticsearchSinkTask.java:130)
```

So it seems as though the ES connector is unable to handle tombstones today, so it is reasonable that we could change the behavior to *drop* such records by default rather than treat them as deletes. I agree that nobody could be using tombstones today, but not everyone would want to delete docs from their indexes, and doing so should be an opt-in.

**Comment by zmays** [comment]
Any movement on this? I'm trying to use debezium to sync postgresql with elasticsearch and propagating deletes would be great.

**Comment by zmays** [comment]
ah wait, is that what this config values is supposed to be for? https://github.com/confluentinc/kafka-connect-elasticsearch/blob/master/src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java#L131

**Comment by gunnarmorling** [comment]
It seems this has been resolved as of PR https://github.com/confluentinc/kafka-connect-elasticsearch/pull/165 and the [4.0.1 release](https://docs.confluent.io/current/connect/connect-elasticsearch/docs/changelog.html#version-4-0-1)?

**Comment by levzem** [comment]
This has been resolved by PR #165 and consequently closing
